pkg/frontend/transport/handler.go

```diff
@@ -252,6 +252,52 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (f *Handler) parseURLValues(r *http.Request, stats *querier_stats.QueryStats, values url.Values) {
+	query := values.Get("query")
+	step := values.Get("step")
+	start := values.Get("start")
+	end := values.Get("end")
+	ts := values.Get("time")
+	var (
+		tsInt, startInt, endInt, stepInt int64
+		err                              error
+	)
+	stats.AddQuery(query)
+	if len(step) == 0 { // instant query
+		if len(ts) > 0 {
+			tsInt, err = util.ParseTime(ts)
+			if err != nil {
+				level.Error(util_log.WithContext(r.Context(), f.log)).Log("msg", "failed to parse time", "err", err)
+			}
```

Comment:
Shouldn't we return after logging the error?

Reply:
yes. will add return there.